File: api/patient-record-service/src/main/java/org/emr/patientrecordservice/models/Visit.java
package org.emr.patientrecordservice.models;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
public class Visit {
    private LocalDate date;
    private String reason;
    private String notes;
    private long healthcareProviderId;
}

File: client/src/models/Visit.js
class Visit {
    constructor(date,reason,notes,healthcareProviderId){
        this.date = date
        this.reason = reason
        this.notes = notes
        this.healthcareProviderId = healthcareProviderId
    }

    static createWithDefaults(){
        return new Visit('','','',null)
    }

    static fromJson(json){
        return new Visit(
            json.date,
            json.reason,
            json.notes,
            json.healthcareProviderId
        )
    }

    isUpcoming(){
        const today = new Date()
        const apptDate = new Date(this.date)

        return today < apptDate
    }
}

export default Visit